File: SharedUtils/src/main/java/com/braintreepayments/api/sharedutils/HttpResponseParser.java
package com.braintreepayments.api.sharedutils;

import androidx.annotation.RestrictTo;

import java.net.HttpURLConnection;

@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public interface HttpResponseParser {
    String parse(int responseCode, HttpURLConnection connection) throws Exception;
}

File: SharedUtils/src/main/java/com/braintreepayments/api/sharedutils/HttpResponseParser.kt
package com.braintreepayments.api.sharedutils

import androidx.annotation.RestrictTo
import java.net.HttpURLConnection

@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
interface HttpResponseParser {
    @Throws(Exception::class)
    fun parse(responseCode: Int, connection: HttpURLConnection): String
}
